src/main/java/br/com/sbs/cubatech/report/ReportDao.java

```diff
@@ -0,0 +1,58 @@
+package br.com.sbs.cubatech.report;
+import br.com.sbs.cubatech.connection.ConnectionFactory;
+import br.com.sbs.cubatech.connection.DaoException;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReportDao {
+
+    private Connection connection;
+
+    @Deprecated
+    public ReportDao() {
+    }
+
+    public ReportDao(Connection connection){
+        this.connection = connection;
+    }
+
+    public List<ReportDTO> getReportData() throws SQLException {
+
+        String querySelect = "SELECT c.id CourseId, c.name CourseName, c.timeToFinishInHours TimeOfCourse, s.id SubCategoryId, s.name SubCategoryName " +
+                "FROM  subCategory s, course c " +
+                "WHERE c.courseVisibility = 'PUBLIC' AND c.subCategory_id = s.id " +
```

Comment:
Era melhor usar aquele JOIN nesse query, evitando o produto cartesiano. No MySQL, eu **acho** que ele não otimiza.

Dá pra usar aquela String multinhas do Java 16 também.

```suggestion
                FROM  subCategory s
                JOIN Course c on c.subCategory_id = s.id
                WHERE c.courseVisibility = 'PUBLIC'
```